src/platform/intel/cavs/lib/clk.c

```diff
@@ -107,26 +120,24 @@ void platform_clock_on_waiti(void)
 		 * so they can be switched without delay on wake up.
 		 */
 		select_cpu_clock(CPU_LPRO_FREQ_IDX, false);
-		set_cpu_current_freq_idx(CPU_LPRO_FREQ_IDX);
 	}
 }
 
-static inline void clock_platform_update_active_freq_idx(int freq_idx)
-{
-	int *uncached_freq_idx = cache_to_uncache(&active_freq_idx);
-
-	*uncached_freq_idx = freq_idx;
-}
 #else
-static inline void clock_platform_update_active_freq_idx(int freq_idx)
+/*
+ * This is call from public API, there is no clock switch for core waiti, so
+ * value of value frequency index in 'active' state doesn't need to be saved in
+ * any additional variable and restored in future.
+ */
+static inline void set_cpu_current_freq_idx(int freq_idx, bool release_unused)
```

Comment:
Are we expecting the public API users to know the mappings between frequency index and frequency in Hz ?

Reply:
In `sof/lib/clk.h` there is function 
```
void clock_set_freq(int clock, uint32_t hz) {
...
idx = clock_get_nearest_freq_idx(clk_info->freqs, clk_info->freqs_num, hz);
...
clk_info->set_freq(clock, idx);
...
}
```